flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -606,25 +606,33 @@ runBashJavaUtilsCmd() {
     local class_path=$2
     local conf_dir=$3
 
-    local output="`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} 2> /dev/null`"
+    local output=`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} | tail -n 1`
     if [[ $? -ne 0 ]]; then
-        echo "[ERROR] Cannot run BashJavaUtils to execute command ${cmd}."
+        echo "[ERROR] Cannot run BashJavaUtils to execute command ${cmd}." 1>&2
+        # Print the output in case the user redirect the log to console.
+        echo $output 1>&2
         exit 1
     fi
 
-    echo ${output}
+    if ! [[ $output =~ ^BASH_JAVA_UTILS_EXEC_RESULT:.* ]]; then
+        echo "[ERROR] The last line of the output does not start with the certain prefix." 1>&2
```

Comment:
I would put it in the following way:
`[ERROR] Unexpected result: $output`
`[ERROR] The last line of the BashJavaUtils outputs is expected to be the execution result, following the prefix "BASH_JAVA_UTILS_EXEC_RESULTS: ".`